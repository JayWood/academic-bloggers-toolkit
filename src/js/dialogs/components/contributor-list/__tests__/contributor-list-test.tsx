import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';

import ManualDataStore from 'stores/data/manual-data-store';
import ContributorList from '..';

const setup = (citationType: CSL.ItemType = 'webpage') => {
    const store = new ManualDataStore(citationType);
    const component = shallow(
        <ContributorList
            citationType={store.citationType}
            people={store.people}
        />,
    );
    return {
        component,
        instance: component.instance() as ContributorList,
    };
};

describe('<ContributorList />', () => {
    const BASELINE = toJSON(setup().component);
    it('should match baseline snapshots', () => {
        expect(BASELINE).toMatchSnapshot();
    });
    it('should add and remove contributors', () => {
        const { component, instance } = setup();
        const addButton = component.find('Button').last();
        addButton.simulate('click');
        expect(toJSON(component, { noKey: true })).toMatchDiffSnapshot(
            BASELINE,
        );

        instance.remove({ currentTarget: { dataset: { index: '0' } } } as any);
        component.update();
        expect(toJSON(component, { noKey: true })).toMatchDiffSnapshot(
            BASELINE,
        );
    });
    it('should throw an error when contributor index is not a number', () => {
        const { instance } = setup();
        expect(() =>
            instance.remove({
                currentTarget: { dataset: { index: 'uh-oh!' } },
            } as any),
        ).toThrowError(TypeError);
    });
});
