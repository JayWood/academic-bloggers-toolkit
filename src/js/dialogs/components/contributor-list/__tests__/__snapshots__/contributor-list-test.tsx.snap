// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContributorList /> should add and remove contributors 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -2,30 +2,10 @@
    <h2
      className=\\"heading\\"
    >
      Contributors
    </h2>
-   <Contributor
-     contributor={
-       Object {
-         \\"family\\": \\"\\",
-         \\"given\\": \\"\\",
-         \\"literal\\": \\"\\",
-         \\"type\\": \\"author\\",
-       }
-     }
-     contributorTypes={
-       Array [
-         Object {
-           \\"label\\": \\"Author\\",
-           \\"type\\": \\"author\\",
-         },
-       ]
-     }
-     index={0}
-     onRemove={[Function res]}
-   />
    <div
      className=\\"btnRow\\"
    >
      <Button
        flat={true}"
`;

exports[`<ContributorList /> should add and remove contributors 2`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`<ContributorList /> should match baseline snapshots 1`] = `
<React.Fragment>
  <h2
    className="heading"
  >
    Contributors
  </h2>
  <div
    className="btnRow"
  >
    <Button
      flat={true}
      label="Add contributor"
      onClick={[Function]}
      type="button"
    />
  </div>
</React.Fragment>
`;
